src/math/integerhistogram1d.h

```diff
@@ -0,0 +1,72 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// Copyright (c) 2023 Team Dissolve and contributors
+
+#pragma once
+
+#include "keywords/optionalint.h"
+#include "math/data1d.h"
+#include "math/sampleddouble.h"
+#include <map>
+
+// Pure Integer Histogram
+class IntegerHistogram1D
+{
+    public:
+    IntegerHistogram1D();
+    ~IntegerHistogram1D() = default;
+    IntegerHistogram1D(const IntegerHistogram1D &source);
+    // Clear data
+    void clear();
+
+    /*
+     * Histogram Data
+     */
+    private:
+    // Minimum value for binning
+    std::optional<int> minimum_{0};
+    // Maximum value for data (hard right-edge of last bin, adjusted to match bin width if necessary)
+    std::optional<int> maximum_;
+    // Number of values binned over all bins
+    long int nBinned_;
+    // Number of points missed (out of bin range)
+    long int nMissed_;
+    // Accumulated data
+    Data1D accumulatedData_;
+    // Map of histogram bin values to instantaneous counts and accumulated averages
+    std::map<int, long int> raw_;
+    std::map<int, SampledDouble> averages_;
+
+    private:
+    // Update accumulated data
+    void updateAccumulatedData();
+    const std::pair<Data1D, std::optional<int>> createDisplayData();
```

Comment:
Just need a comment for the function
```suggestion
    // Create display data object covering extents of current bins
    const std::pair<Data1D, std::optional<int>> createDisplayData();
```